project/BuildPlugin.scala

```diff
@@ -0,0 +1,357 @@
+
+    import BuildAutoImported.{ ZincAlternativeCacheName, ZincAlternativeCacheDir }
+    val AlternativeLocalResolver: Resolver =
+      Resolver.file(ZincAlternativeCacheName, ZincAlternativeCacheDir)(Resolver.ivyStylePatterns)
+
+    // Naive way of implementing a filter to remove repeated resolvers.
+    def removeRepeatedResolvers(rs: Seq[Resolver]): Seq[Resolver] = rs.distinct
+  }
+
+  object BuildCommands {
+    import sbt.{ Command, State }
+    import BuildAutoImported.bridgeScalaVersions
+    def crossTestBridges(bridge: Project): Command = {
+      Command.command("crossTestBridges") { (state: State) =>
+        (bridgeScalaVersions.flatMap { (bridgeVersion: String) =>
+          // Note the ! here. You need this so compilerInterface gets forced to the scalaVersion
+          s"++ $bridgeVersion!" :: s"${bridge.id}/test" :: Nil
+        }) ::: (s"++ ${Dependencies.scala212}!" :: state)
+      }
+    }
+
+    def publishBridgesAndSet(bridge: Project, interface: Project, apiInfo: Project): Command = {
+      Command.args("publishBridgesAndSet", "<version>") { (state, args) =>
+        require(args.nonEmpty, "Missing Scala version argument.")
+        val userScalaVersion = args.mkString("")
+        s"${interface.id}/publishLocal" :: bridgeScalaVersions.flatMap { (v: String) =>
+          s"++ $v!" :: s"${apiInfo.id}/publishLocal" :: s"${bridge.id}/publishLocal" :: Nil
+        } ::: s"++ $userScalaVersion!" :: state
+      }
+    }
+
+    def publishBridgesAndTest(bridge: Project, interface: Project, apiInfo: Project): Command = {
+      Command.args("publishBridgesAndTest", "<version>") { (state, args) =>
+        require(args.nonEmpty, "Missing arguments to publishBridgesAndTest.")
+        val version = args mkString ""
+        val bridgeCommands: List[String] = bridgeScalaVersions.flatMap { (v: String) =>
+          s"++ $v" :: s"${apiInfo.id}/publishLocal" :: s"${bridge.id}/publishLocal" :: Nil
+        }
+        s"${interface.id}/publishLocal" ::
+          bridgeCommands :::
+          s"++ $version" ::
+          s"zincRoot/scalaVersion" ::
+          s"zincRoot/test" ::
+          s"zincRoot/scripted" ::
+          state
+      }
+    }
+
+    val release: Command =
+      Command.command("release")(st => "clean" :: "+compile" :: "+publishSigned" :: "reload" :: st)
+
+    def runBenchmarks(benchmarkProject: Project): Command = {
+      val dirPath = BuildAutoImported.benchmarksTestDir.getAbsolutePath
+      val projectId = benchmarkProject.id
+      val runPreSetup = s"$projectId/run $dirPath"
+      val runBenchmark = s"$projectId/jmh:run -p _tempDir=$dirPath -prof gc"
+      val tearDownResources = s"$projectId/tearDownBenchmarkResources"
+      Command.command("runBenchmarks")(st => runPreSetup :: runBenchmark :: tearDownResources :: st)
+    }
+
+    def all(bridge: Project, interface: Project, apiInfo: Project, bench: Project): Seq[Command] = {
```

Comment:
Could you make these `ProjectReference` please? We don't need the actual project, just a reference to it.

Reply:
We cannot because we need `id`, and `id` is not defined in `ProjectReference`.

Reply:
```scala
implicit def projectToRef(p: Project): ProjectReference = LocalProject(p.id)
```

If only this returned `LocalProject` instead of `ProjectReference`..

Reply:
@dwijnand Yeah... can maybe this new method be added in sbt 1.x? I think it's bincompat:
  
```scala
@ class A 
class B edefined class A
@ class B extends A 
defined class B
@ implicit def stringToA(s: String): A = new A 
defined function stringToA
@ implicit def stringToB(s: String): B = new B 
defined function stringToB
```

Reply:
yep, I think we should consider it.